dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/RumEventConsumer.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.webview
+
+import com.datadog.android.core.internal.CoreFeature
+import com.datadog.android.core.internal.persistence.DataWriter
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.domain.RumContext
+import com.datadog.android.rum.internal.domain.scope.RumApplicationScope
+import com.datadog.android.rum.internal.domain.scope.RumSessionScope
+import com.datadog.android.rum.internal.monitor.DatadogRumMonitor
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.LongTaskEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
+import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+
+internal class RumEventConsumer(
+    private val dataWriter: DataWriter<Any>,
+    private val timeProvider: TimeProvider
+) {
+
+    private var currentOffset = timeProvider.getServerOffsetMillis()
+
+    fun consume(event: JsonObject, eventType: String) {
+        if (!isSessionDropped()) {
+            val mappedEvent = map(event, eventType)
+            if (mappedEvent != null) {
+                dataWriter.write(mappedEvent)
+            }
+        }
+    }
+
+    private fun map(event: JsonObject, eventType: String): JsonObject? {
+        val rumContext = GlobalRum.getRumContext()
+
+        return try {
+            when (eventType) {
+                VIEW_EVENT_TYPE -> {
+                    currentOffset = timeProvider.getServerOffsetMillis()
+                    ViewEvent.fromJson(event.toString()).mapToMobile(rumContext, currentOffset)
```

Comment:
An alternative approach could be to not parse with out specialised types (I agree, it might surface many compatibility issues), but just apply patch to the `JSONObject`, something this:
```kotlin
fun mapViewEvent(browserView: JSONObject, context: RumContext, timeOffset: Long): JsonObject? {
   try {
      browserView.replace(keyPath: "application.id", with: context.application.id)
      browserView.replace(keyPath: "session.id", with: context.session.id)
      browserView.replace(keyPath: "service", with: Core.serviceName)
      browserView.replace(keyPath: "view.date", with: browserView["view"]["date"] + offset)
      return browserView // success
   } catch {
      // notify internal error
      return null // failure
   }
}
```
This way we don't rely on schema validation, but we limit the expected interface to only values we need to modify. This won't break on future schema changes unless it's a change to these core fields.

Reply:
was thinking exactly to the same approach but this will mean that we will not check if the events format is correct. I would rather go with the simplified approach and only replace the required properties in the JSON object than to have all these transformations.